app/src/main/java/lu/geoportail/map/LuxTileServer.kt

```diff
@@ -127,21 +312,31 @@ class LuxTileServer(
 
     private val getMbTile =
         HttpServerRequestCallback { request: AsyncHttpServerRequest, response: AsyncHttpServerResponse ->
-            var layer = "road"
-            if (request.query.getString("layer") == "topo") layer = "topo"
+            var layer = request.query.getString("layer") ?: "omt-geoportail"
             val z: Int = request.query.getString("z").toInt()
             val x: Int = request.query.getString("x").toInt()
             var y: Int = request.query.getString("y").toInt()
 
             response.headers.add("Access-Control-Allow-Origin","*")
 
+            if (!db.containsKey(layer)) {
+                try {
+                    db[layer] = openDB(layer)
+                }
+                catch (e: SQLiteCantOpenDatabaseException) {
+                    response.code(404)
```

Comment:
500 error code instead? this is a critical error.

Reply:
it's rather a file not found exception when the correct layer is not available. But It would be possible to check for file existence before trying to open the DB